changelog/src/main/kotlin/com/ritense/valtimo/changelog/service/ChangelogDeployer.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ *  Copyright 2015-2023 Ritense BV, the Netherlands.
+ *
+ *  Licensed under EUPL, Version 1.2 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" basis,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.ritense.valtimo.changelog.service
+
+import com.ritense.valtimo.changelog.domain.ChangesetDeployer
+import mu.KotlinLogging
+import org.springframework.boot.context.event.ApplicationReadyEvent
+import org.springframework.context.event.EventListener
+import org.springframework.transaction.annotation.Transactional
+
+@Transactional
+class ChangelogDeployer(
+    private val changelogService: ChangelogService,
+    private val changesetDeployers: List<ChangesetDeployer>,
+) {
+
+    @EventListener(ApplicationReadyEvent::class)
+    fun deployAll() {
+        logger.info { "Running deployer" }
+
+        changesetDeployers.forEach { it.before() }
+
+        changesetDeployers
+            .flatMap { changesetDeployer ->
+                changelogService.loadResources(changesetDeployer.getPath()).map { Pair(changesetDeployer, it) }
```

Comment:
I don't think there's a need to do this in a flatmap. The order is already written down with the `@Order()` annotation. So instead you could to something like:

```
.forEach { (changesetDeployer) ->
    val changelogs = changelogService.loadResources(changesetDeployer.getPath())
   // for each changelog, import the file
}
```